components/protocol-wasm/src/lib.rs

```diff
@@ -0,0 +1,213 @@
+use chrono::{DateTime, Utc};
+use base64::{Engine as _, engine::{self, general_purpose}};
+use std::str::FromStr; // 0.4.15
+
+use bindings::protocol;
+struct Component;
+
+pub use semver::{Version, VersionReq};
+
+use anyhow::Error;
+use anyhow::anyhow;
+
+use warg_protocol::{
+  package,
+  proto_envelope::{ProtoEnvelope, ProtoEnvelopeBody}, 
+  SerdeEnvelope,
+  registry::{MapCheckpoint, RecordId, LogId, LogLeaf, MapLeaf},
+};
+use warg_crypto::{signing, Decode, hash::{Hash, Sha256, HashAlgorithm, DynHash}};
+use warg_transparency::{log::LogProofBundle, map::MapProofBundle};
+use warg_api::proof::ProofError;
+
+/// Represents information about a registry package.
+#[derive(Debug, Clone)]
+// #[serde(rename_all = "camelCase")]
+pub struct PackageInfo {
+    /// The name of the package.
+    pub name: String,
+    /// The last known checkpoint of the package.
+    // #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub checkpoint: Option<String>,
+    // pub checkpoint: Option<String>,
+    /// The current validation state of the package.
+    // #[serde(default)]
+    pub state: package::Validator,
+}
+
+impl PackageInfo {
+    /// Creates a new package info for the given package name and url.
+    pub fn new(name: impl Into<String>, // checkpoint: protocol::MapCheckpoint
+    ) -> Self {
+        Self {
+            name: name.into(),
+            checkpoint: None,
+            state: package::Validator::default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct MyBody(protocol::ProtoEnvelopeBody);
```

Comment:
What's this newtype for?

Reply:
Oh yeah sorry, relic of just getting things to work.  I renamed to `ProtoBody`